File: Python/GoとPythonの違い/データ構造/dict/for.py
d = {"apple":2, "banana": 3, "orange": 1}

# keyを出力する  d.keys()でも同じ
for a in d:
    print(a)

print("*" * 20)

# key, valueを出力する
for a, b in d.items():
    print(a, b)

print("*" * 20)

# valueを出力する
for a in d.values():
    print(a)

File: Python/GoとPythonの違い/データ構造/dict/for.go
package main

import (
	"fmt"
)


func main(){
   d := map[string]int{"apple": 2, "banana":3, "orange": 1} 

   for k1 := range d{
      fmt.Println(k1)
   }

   fmt.Println("*****************")

   for k2, v2 := range d{
      fmt.Println(k2, v2)
   }

   fmt.Println("*****************")

   for _, v3 := range d{
      fmt.Println(v3)
   }
}
